### PR TITLE
Feat/myc 55 - Search functionality

### PR DESCRIPTION
**Pull request by maplonki**
This adds the location request portion of the search and displays the user location on the map. 

Will update once the backend portion is finished

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,6 +38,12 @@ captures/
 
 # IntelliJ
 *.iml
+.idea/codeStyles/codeStyleConfig.xml
```
Comment (codecakes) [inline .gitignore:41]: can we not do just `.idea/` ?

Comment (jumaallan) [reply .gitignore:41]: +1 on this

**Review by jumaallan — APPROVED** [review]

**Comment by codecakes** [comment]
@jumaallan @maplonki fastlane android test CI is breaking;

**Comment by jumaallan** [comment]
I noticed

Issue is the key.properties file 

@maplonki - you can pass the config via Fastlane. Let me know if you can set that up

**Review by codecakes — APPROVED** [review]
We should have a precommit hook that ktlints automatically:
https://medium.com/binar-academy/android-kotlin-code-quality-check-using-detekt-and-git-hooks-pre-push-mac-and-linux-556c111bc71b

**Comment by maplonki** [comment]
> I noticed
> 
> Issue is the key.properties file
> 
> @maplonki - you can pass the config via Fastlane. Let me know if you can set that up

@jumaallan can you help out with that?

**Comment by codecakes** [comment]
> > I noticed
> > Issue is the key.properties file
> > @maplonki - you can pass the config via Fastlane. Let me know if you can set that up
> 
> @jumaallan can you help out with that?

@jumaallan you can push this commit

**Comment by codecakes** [comment]
any updates here?

**Comment by jumaallan** [comment]
> any updates here?

Let me fix that today

**Comment by jumaallan** [comment]
> > any updates here?
> 
> Let me fix that today

@codecakes @maplonki 

I've replaced the implementation that reads from keys.properties file, and replaced with one that reads from gradle.properties - this can be done in two ways:

1. on the local gradle.properties 
2. on the global gradle.properties

I'll document on the README, how to do that. 

I have also updated Fastlane to read the Maps API Key, from Github Secrets when doing a build - @codecakes - Kindly create a secret with the name MAPS_DEBUG_KEY, and paste the debug API Key - builds should now work with that

**Comment by codecakes** [comment]
> > > any updates here?
> > 
> > 
> > Let me fix that today
> 
> @codecakes @maplonki
> 
> I've replaced the implementation that reads from keys.properties file, and replaced with one that reads from gradle.properties - this can be done in two ways:
> 
>     1. on the local gradle.properties
> 
>     2. on the global gradle.properties
> 
> 
> I'll document on the README, how to do that.
> 
> I have also updated Fastlane to read the Maps API Key, from Github Secrets when doing a build - @codecakes - Kindly create a secret with the name MAPS_DEBUG_KEY, and paste the debug API Key - builds should now work with that

Thanks @jumaallan ; could you ping me our group the api key?

**Review by codecakes — CHANGES_REQUESTED** [review]
@jumaallan is the linter used in the ide adding blank line EOF?

**Comment by jumaallan** [comment]
> > > > any updates here?
> > > 
> > > 
> > > Let me fix that today
> > 
> > 
> > @codecakes @maplonki
> > I've replaced the implementation that reads from keys.properties file, and replaced with one that reads from gradle.properties - this can be done in two ways:
> > ```
> > 1. on the local gradle.properties
> > 
> > 2. on the global gradle.properties
> > ```
> > 
> > 
> > I'll document on the README, how to do that.
> > I have also updated Fastlane to read the Maps API Key, from Github Secrets when doing a build - @codecakes - Kindly create a secret with the name MAPS_DEBUG_KEY, and paste the debug API Key - builds should now work with that
> 
> Thanks @jumaallan ; could you ping me our group the api key?

I don't have the API Key, probably @maplonki has the keys

**Comment by jumaallan** [comment]
> @jumaallan is the linter used in the ide adding blank line EOF?

I don't think so - I will setup the following:
1. https://github.com/JLLeitschuh/ktlint-gradle
2. https://github.com/detekt/detekt
3. https://github.com/diffplug/spotless

I'll do a separate PR for that

**Comment by codecakes** [comment]
> > @jumaallan is the linter used in the ide adding blank line EOF?
> 
> I don't think so - I will setup the following:
> 
>     1. https://github.com/JLLeitschuh/ktlint-gradle
> 
>     2. https://github.com/detekt/detekt
> 
>     3. https://github.com/diffplug/spotless
> 
> 
> I'll do a separate PR for that

thanks alot!@

**Comment by jumaallan** [comment]
Added some TODOs, and dummy code to kill the linter warnings - FYI

@maplonki @codecakes 

We need to merge all PRs to develop, and rework from there

**Comment by jumaallan** [comment]
image

@maplonki can you merge this? or @codecakes  - I am not able to merge the PR

**Review by codecakes — APPROVED** [review]
Use smart commits if working on a JIRA user ticket: https://support.atlassian.com/bitbucket-cloud/docs/use-smart-commits/

**Pull request merged**